core/src/main/java/gyro/core/validation/ConflictsWithValidator.java

```diff
@@ -0,0 +1,36 @@
+package gyro.core.validation;
+
+import gyro.core.resource.Diffable;
+import gyro.core.resource.DiffableField;
+import gyro.core.resource.DiffableType;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.stream.Stream;
+
+public class ConflictsWithValidator implements Validator<ConflictsWith> {
+    @Override
+    public boolean isValid(Diffable diffable, ConflictsWith annotation, Object value) {
+        DiffableType<Diffable> diffableType = DiffableType.getInstance(diffable);
+
+        return  (isValueEmpty(value) || Stream.of(annotation.value()).allMatch(o -> isFieldEmpty(o, diffableType, diffable)));
+    }
+
+    @Override
+    public String getMessage(ConflictsWith annotation) {
+        return String.format("Cannot be set when any of '%s' is set."
```

Comment:
I'd tweak this to use the word "field" and account for plural. Maybe`Cannot be set when any of the following field(s) are set: [%s]`?